lib/spack/docs/build_systems/gopackage.rst

```diff
@@ -0,0 +1,164 @@
+.. Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+.. _gopackage:
+
+-----------
+GoPackage
+-----------
+
+There are a couple of things to be aware of about applications built
+with Go:
+
+* Go-based projects use a variety of build tools: simple invocations
+  of ``go build``, custom go "scripts" that leverage ``go run``, Make,
+  and *etc...*.
+
+* Go's dependency management story has evolved from an initial
+  do-it-yourself attitude through a small number of commonly used
+  tools to an officially supported approach (Go "modules").  The Go
+  team provides an `introduction to Go modules
+  <https://blog.golang.org/using-go-modules>`_ and maintains a site
+  that tracks `the gory details
+  <https://github.com/golang/go/wiki/Modules>`_.
+
+  Go modules are the future, ``GOPATH`` will likely be deprecated in
+  Go 1.14.
+
+Not surprisingly, the best approach to packaging a Go-based project
+for Spack depends on the details of the project.
+
+The ``GoPackage`` base class is intended to make it easy to package
+simple projects.  Tools & techniques that it uses may also be useful
+when using other build system base classes (e.g. ``MakefilePackage``),
+in particular:
+
+- Use the package's ``setup_build_environment`` to set
+  ``GO111MODULE=on`` to enable module mode and set
+  ``GOFLAGS="-mod-vendor"`` to enable vendoring, which prevents ``go``
```

Comment:
```suggestion
  ``GOFLAGS=-mod-vendor`` to enable vendoring, which prevents ``go``
```
Quotes shouldn't be needed here, as nothing is ever run through a Bash shell, only a Python shell.

Reply:
fixed